src/main/java/com/michelin/kafkactl/util/ResourceDependencySorter.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.michelin.kafkactl.util;
+
+import java.util.*;
+
+public class ResourceDependencySorter {
+
+    public static List<String> sortResourceNamesByDependencies(
+            Set<String> resourceNames, Map<String, Set<String>> dependencies) {
+        List<String> sorted = new ArrayList<>();
+        Set<String> visited = new HashSet<>();
+        Set<String> visiting = new HashSet<>();
+        for (String name : resourceNames) {
+            visit(name, dependencies, visited, visiting, sorted);
+        }
+        return sorted;
+    }
+
+    private static void visit(
```

Comment:
I found this function a little bit tricky. Some suggestion (to be discussed) to simplify it:
- `sorted` and `visited` are always updated at the same time. Do we really need 2 separate lists?

- merge `visiting.contains` and `visiting.add`. To be tested, but the `add` function return `false` if it contains the element already, so I think we can just add or throw exception in one line:

```java
if (!visiting.add(resource)) {
    throw new IllegalStateException("Cyclic dependency detected");
}
```

- the `for` loop can be rewrite like so:

```java
deps.getOrDefault(resource, Set.of())
        .forEach(dep -> visit(dep, deps, visited, visiting, result));
```

Theses changes are suggestions and need to be tested carefully